app/user.go

```diff
@@ -1853,3 +1871,82 @@ func (a *App) UpdateOAuthUserAttrs(userData io.Reader, user *model.User, provide
 
 	return nil
 }
+
+// UsersAreTeamGroupMembers verifies that all of the users are members of groups associated to the team. If one or
+// more users are not associated then a model.InvalidTeamMembersError is returned containing the user ids.
+func (a *App) UsersAreTeamGroupMembers(userIDs []string, team *model.Team) error {
```

Comment:
The naming makes me expect that this function is going to return a `bool` and maybe the collection of invalid members rather than an error, sounds more like a check instead of an enforcement, that is going to leave the responsibility to decide if there is an error or not to the caller.

If we want to return an error, I'd propose something like `AssertUsersAreTeamGroupMembers` 2/5

Reply:
You make a good point and I agree that the names should change. 

As for what it should change to `AssertUsersAre[Team|Channel]GroupMembers` is one option, also `EnsureUsersAre[Team|Channel]GroupMembers` or `CheckUsersAre[Team|Channel]GroupMembers` follow patterns used in the Go standard library for functions that return only `error`.

Another approach is to change the name *and* signature of the method to have it return a slice of non-team/channel-group members, for example:
```go
NonTeamGroupMembers(userIDs []string, team *model.Team) []string
```
... and then instead of the caller checking for an error they check for an empty slice. This would make it more semantically correct if one were using the method to *actually retrieve* non-members, in which case one wouldn't consider it to be an error case.

For the name change I don't have a preference between assert/check/ensure but I'm leaning towards the name and signature change option for semantic flexibility. 

What are your thoughts?

Reply:
As for your first name proposals, I really like the use of `Ensure`, I think is quite explanatory and it's great that is part of the Go standard library.

I agree on the name + signature change, and I think that this specific method can be useful in the future, so I think it's a great solution for this! I'm only missing a verb from that method name, I'd use something like `GetNonTeamGroupMembers`.